hazelcast/src/main/java/com/hazelcast/map/impl/PartitionContainer.java

```diff
@@ -239,6 +239,28 @@ public void setLastCleanupTimeCopy(long lastCleanupTimeCopy) {
         this.lastCleanupTimeCopy = lastCleanupTimeCopy;
     }
 
+    /**
+     * Cleans up the container's state if the enclosing partition is migrated
+     * off this member. Whether cleanup is needed is decided based on the
+     * provided {@code replicaIndex}.
+     *
+     * @param replicaIndex The replica index to use for deciding per map whether
+     *                     cleanup is necessary or not
+     */
+    final void cleanUpOnMigration(int replicaIndex) {
+        MapServiceContext mapServiceContext = mapService.getMapServiceContext();
+        NodeEngine nodeEngine = mapServiceContext.getNodeEngine();
+
+        mapServiceContext.getMapContainers().keySet().stream()
+                .filter(mapName -> replicaIndex == -1
+                        || replicaIndex > nodeEngine.getConfig().findMapConfig(mapName).getTotalBackupCount())
```

Comment:
Can we do this filtering over record-store? Asking this since it will be faster if we can do, we eliminate `findMapConfig ` complexity. 

Like this: `recordStore.getMapContainer().getTotalBackupCount()`

Reply:
I was looking for something like this, but didn't check `getAllRecordStores()` and what's behind it, thanks for the hint. Sure, I change the filtering.

Reply:
Addressed in https://github.com/hazelcast/hazelcast/pull/21517/commits/fd27531ed61a51ed6d123a7b4296a2816dbb5842